### ISSUE TITLE
[Bug]: Unlimited money bug by selling unlocked items

### ISSUE THREAD

**Issue by Jul1aK0wal1k** [igorkis-scrts/A3-Antistasi-Plus#66]
### Describe the bug

You can sell unlocked items to the arms dealer making it a source of unlimited cash. I expected to not be able to sell unlocked items when selling stuff to the arms dealer.

### How to reproduce

1. Find the arms dealer
2. Sell any unlocked weapon or magazine
3. Unlimited profit 

### Version

1.7.5

### Have you altered the code?

No

### Map

Island Panthera

### What server?

Private dedicated server with Headless client(s)

### Time bug occured (Server time/UTC)

23:00

### Mods

```html
Server side mods 
https://steamcommunity.com/sharedfiles/filedetails/?id=2624565981
```

**Comment by igorkis-scrts** [comment]
Can't reproduce this behaviour as some forms of protection are already there - unlocked items are removed from stock with each arsenal tick and there is also a check if seller are trying to sell something from unlocked items list on each sell button push. 
Maybe there is some rare case though.

**Comment by igorkis-scrts** [comment]
Yeah, investigated it further and unlock check basically didn't worked due to nested array.
Thanks for report, will be fixed in the upcoming 1.8.
